source/server/server.cc

```diff
@@ -234,12 +234,22 @@ void InstanceUtil::loadBootstrapConfig(envoy::config::bootstrap::v3::Bootstrap&
                          "should be non-empty");
   }
 
+  absl::optional<MessageVersion> version;
+
+  if (options.bootstrapVersion()) {
```

Comment:
Yeah, I agree this is a bit ungraceful. I think the alternative is we supply to `loadFromJson` etc. the type URL of the bootstrap that we know exists on the filesystem, e.g. `envoy.config.bootstrap.v2`. We can do that mapping here.

Then, we load JSON specifically into a message that we build from reflection with that type. We then have a v2 bootstrap message. Following this, we do a `VersionConverter::upgrade` to the v3 bootstrap Envoy is using internally. LMK if you need more details on this, we have some examples of each of these things around.

Reply:
That makes sense to me, it's along the lines of what I had in mind as well for a way doing this by bypassing the whole `withApiBoosting` helper. I'll give that a go and see how it goes